src/main/java/me/crafter/mc/lockettepro/BlockPlayerListener.java

```diff
@@ -227,23 +232,37 @@ public void onAttemptBreakSign(BlockBreakEvent event) {
     }
 
     //protect sign from being changed
-    @EventHandler(priority = EventPriority.HIGH, ignoreCancelled = true)
-    public void onAttemptChangeLockerSign(SignChangeEvent event){
+    @EventHandler(priority = EventPriority.LOW, ignoreCancelled = true)
+    public void onAttemptChangeLockerSign(SignChangeEvent event) {
         Block block = event.getBlock();
-        if(LocketteProAPI.isLockSign(block)||LocketteProAPI.isAdditionalSign(block)){
-            Sign sign = (Sign)block.getState();
+        if (LocketteProAPI.isLockSign(block) || LocketteProAPI.isAdditionalSign(block)) {
+            Sign sign = (Sign) block.getState();
             sign.setWaxed(true);
             sign.update();
             event.setCancelled(true);
+
+            // langua added this, however it doesn't seem on brant since
+            // a.) everything works as intended
+            // b.) smoke particles are used nowhere else
+            //block.getWorld().spawnParticle(Particle.SMOKE_NORMAL, block.getLocation(), 5);
         }
-        block.getWorld().spawnParticle(Particle.SMOKE_NORMAL,block.getLocation(),5);
     }
-    @EventHandler(priority = EventPriority.HIGH,ignoreCancelled = true)
-    public void onAttemptBreakWaxedLockerSign(PlayerInteractEvent event){
+
+    @EventHandler(priority = EventPriority.HIGH, ignoreCancelled = true)
+    public void onAttemptBreakWaxedLockerSign(PlayerInteractEvent event) {
         Action action = event.getAction();
         Block block = event.getClickedBlock();
-        boolean isAxe = Objects.equals(event.getItem(), new ItemStack(Material.WOODEN_AXE)) || Objects.equals(event.getItem(), new ItemStack(Material.STONE_AXE)) || Objects.equals(event.getItem(), new ItemStack(Material.IRON_AXE)) || Objects.equals(event.getItem(), new ItemStack(Material.GOLDEN_AXE)) || Objects.equals(event.getItem(), new ItemStack(Material.DIAMOND_AXE)) || Objects.equals(event.getItem(), new ItemStack(Material.NETHERITE_AXE));
-        if(action == Action.RIGHT_CLICK_BLOCK&&(LocketteProAPI.isLockSign(block)||LocketteProAPI.isAdditionalSign(block))&&isAxe){
+
+        boolean isAxe = false;
+        if (event.getItem() != null) {
+            switch (event.getItem().getType()) {
+                case WOODEN_AXE, STONE_AXE, IRON_AXE, GOLDEN_AXE, DIAMOND_AXE, NETHERITE_AXE -> isAxe = true;
```

Comment:
You could use the tag: https://hub.spigotmc.org/javadocs/spigot/org/bukkit/Tag.html#ITEMS_AXES

Reply:
Nice one. I was wondering if something like this existed!

Reply:
Ah, I see ^^